tools/lisa-build-asset

```diff
@@ -58,24 +87,70 @@ def list_assets(recipe_dir):
     )
 
 def get_asset_recipe(asset, recipe_dir):
-    return os.path.join(recipe_dir, '{}.recipe'.format(asset))
+    return os.path.join(recipe_dir, f'{asset}.recipe')
+
+def get_recipe_name(asset):
+    return f'{asset}.recipe'
+
+def get_asset_property(asset, property, *args):
+    # Read asset's recipe to get a property value.
+    action = f"printf '%s\\n' ${{{property}[@]}}"
+    val = make(asset, [action], *args)
+    return val[action].splitlines()
+
+@contextmanager
+def make_chroot(asset, arch, recipe_dir, build_dir, arch_chroot_dir):
+    # Create a chroot at arch_chroot_dir as a contextmanager.
+    recipe = get_asset_recipe(asset, recipe_dir)
+    env = get_env(asset, arch, None, recipe, build_dir)
+
+    # List the required Alpine packages.
+    alpine_dependencies = get_asset_property(asset, "ALPINE_BUILD_DEPENDENCIES", arch, None, recipe_dir, build_dir)
+    alpine_dependencies_args = ('-p', ' '.join(map(quote, alpine_dependencies))) if alpine_dependencies else None
+
+    # List the required env variables.
+    used_env_variables = get_used_env_var(get_asset_recipe(asset, recipe_dir), env)
+    used_env_variables_args = ('-k', ' '.join(used_env_variables)) if used_env_variables else None
+
+    # Install the chroot.
+    subprocess.check_call(['sudo', os.path.join(get_alpine_dir(), 'alpine-chroot-install'),
+            '-d', arch_chroot_dir,
+            '-i', build_dir,
+            '-i', recipe_dir,
+            '-a', QEMU_ARCH_NAMES.get(arch),
+            *alpine_dependencies_args,
+            *used_env_variables_args],
+            cwd=build_dir)
 
-def make(asset, actions, arch, toolchain, recipe_dir, build_dir):
+    try:
+        yield arch_chroot_dir
+    finally:
+        # Clean-up the chroot
+        subprocess.check_call([os.path.join(arch_chroot_dir, 'destroy'), '--remove'])
+
+def make(asset, actions, arch, toolchain, recipe_dir, build_dir, arch_chroot_dir=None, native_build=False):
```

Comment:
Is `native_build` parameter needed ? It looks like a check on `arch_root_dir` would achieve the same thing.